java-client/weather-server-example/src/main/java/io/deephaven/client/examples/ui/Ui.java

```diff
@@ -0,0 +1,262 @@
+import io.deephaven.client.impl.BarrageSession;
+import io.deephaven.client.impl.ConsoleSession;
+import io.deephaven.client.impl.SessionImpl;
+import io.deephaven.client.impl.SessionImplConfig;
+import io.deephaven.client.impl.script.Changes;
+import io.deephaven.engine.updategraph.UpdateGraphProcessor;
+import io.deephaven.extensions.barrage.table.BarrageTable;
+import io.deephaven.proto.DeephavenChannel;
+import io.deephaven.util.ExceptionDetails;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.util.concurrent.*;
+
+public class Ui {
+    private JPanel panel1;
+    private JTextField locationField;
+    private JButton addButton;
+    private JTable summaryTable;
+    private JLabel statusLabel;
+    private JTextField hostField;
+    private JButton connectButton;
+    private JCheckBox plaintextCheckBox;
+    private SessionImpl session;
+
+    // Connection related items both for GRPC and Flight.
+    private final ScheduledExecutorService flightScheduler = Executors.newScheduledThreadPool(8);
+    private ManagedChannel managedChannel;
+    private BarrageSupport support;
+
+
+    private BarrageTable statsTable;
+
+    public Ui() {
+        // Start the LTM. This module is responsible for deterministically handling table updates
+        // In other words, it drives the 'ticking' of tables.
+        UpdateGraphProcessor.DEFAULT.start();
+
+        connectButton.addActionListener(this::doConnect);
+        addButton.addActionListener(this::onAdd);
+
+        Runtime.getRuntime().addShutdownHook(new Thread(this::onShutdown));
+    }
+
+    private void onAdd(final ActionEvent ev) {
+        // First we'll just ship the request to the server and wait for the response.
+        // If we get an Ack (true) We'll stuff this thing in the list. This is crude on purpose --
+        // We don't actually care about having complex logic to track places, only that we can submit them.
+        final String place = locationField.getText();
+        if (place == null || place.isEmpty()) {
+            return;
+        }
+
+        try (final ConsoleSession console = session.console("python").get()) {
+            SwingUtilities.invokeLater(() -> statusLabel.setText("Attempting to add " + place));
+            final Changes c = console.executeCode("beginWatch(\"" + place + "\")");
+            if (c.errorMessage().isPresent()) {
+                SwingUtilities.invokeLater(
+                        () -> statusLabel.setText("Error adding " + place + " -> " + c.errorMessage().get()));
+            } else {
+                SwingUtilities.invokeLater(() -> statusLabel.setText("Added " + place));
+            }
+        } catch (ExecutionException | InterruptedException | TimeoutException e) {
+            e.printStackTrace();
```

Comment:
Hmm. At this point, why not just catch Exception/Throwable and log it non-noncommittally like this?